frontend/public/components/monitoring/alert-manager-yaml-editor.tsx

```diff
@@ -20,30 +20,28 @@ const AlertManagerYAMLEditor: React.FC<AlertManagerYAMLEditorProps> = ({
   onCancel = history.goBack,
 }) => {
   const secret: K8sResourceKind = obj;
-  const alertManagerYaml = _.get(secret, ['data', 'alertmanager.yaml']);
-  const initErrorMsg = _.isEmpty(alertManagerYaml)
+  let encodedAlertManagerYaml = _.get(secret, ['data', 'alertmanager.yaml']);
+  const initErrorMsg = _.isEmpty(encodedAlertManagerYaml)
     ? 'Error: alertmanager.yaml not found in Secret "alertmanager-main", in namespace "openshift-monitoring"'
     : null;
 
   const [errorMsg, setErrorMsg] = React.useState(initErrorMsg);
   const [successMsg, setSuccessMsg] = React.useState();
   const [inProgress, setInProgress] = React.useState(false);
-  const [yamlStringData, setYamlStringData] = React.useState(
-    !_.isEmpty(alertManagerYaml) ? Base64.decode(alertManagerYaml) : '',
+  const [alertManagerYamlStr, setAlertManagerYamlStr] = React.useState(
+    !_.isEmpty(encodedAlertManagerYaml) ? Base64.decode(encodedAlertManagerYaml) : '',
   );
 
   const save = (e) => {
     e.preventDefault();
-    if (_.isEmpty(yamlStringData)) {
+    if (_.isEmpty(alertManagerYamlStr)) {
       setErrorMsg('Alertmanager configuration cannot be empty.');
       setSuccessMsg('');
       return;
     }
     setInProgress(true);
-    const patch = [
-      { op: 'replace', path: '/data/alertmanager.yaml', value: Base64.encode(yamlStringData) },
-    ];
-    k8sPatch(SecretModel, secret, patch).then(
+    encodedAlertManagerYaml = Base64.encode(alertManagerYamlStr);
```

Comment:
I think it's better not to reuse `encodedAlertManagerYaml` here.
Also, maybe better for the `Base64.encode` to happen inside `patchAlertManagerConfig` since that encoding should only be necessary when making the API call?

Reply:
I thought of that, but figured I wanted to return an error message if `safeDump/encode` failed, and a specific error msg in `k8sPatch` failed.  I'll look at it again and see if I can do that from within `patchAlertManagerConfig`  -thanks

Reply:
fixed, removed `encodedAlertManagerYaml` method and put code inside of `patchAlertManagerConfig`, see last commit